src/types/integer.rs

```diff
@@ -220,13 +230,34 @@ impl_uint_from_small_signed_int_types!(i8, i16, i32, i64, isize);
 
 impl From<u128> for UInt {
     fn from(value: u128) -> UInt {
-        UInt::BigUInt(BigUint::from(value))
+        UIntData::BigUInt(BigUint::from(value)).into()
     }
 }
 
+macro_rules! impl_int_types_from_uint {
+    ($($t:ty),*) => ($(
+        impl TryFrom<&UInt> for $t {
+            // This allows the caller to reclaim the UInt in the event that conversion fails.
+            type Error = ();
+
+            // If conversion fails, we discard the error. `TryFromIntError` provides no
+            // additional context and cannot be constructed outside of `std`. We'll return
+            // the unit type, `()`.
+            fn try_from(value: &UInt) -> Result<Self, Self::Error> {
+                match &value.data {
+                    UIntData::U64(value) => <$t>::try_from(*value).map_err(|_|()),
+                    UIntData::BigUInt(value) => <$t>::try_from(value).map_err(|_|()),
+                }
+            }
+        }
+    )*)
+}
+
+impl_int_types_from_uint!(i8, i16, i32, i64, i128, u8, u16, u32, u64, u128);
```

Comment:
🗺️ Previously, we only provided a `TryFrom<&UInt>` implementation for `usize` and `i64`. That was more or less acceptable because `UInt` wasn't part of the public API. However, I think it needs to be in a couple of places. For example, a `Decimal::coefficient` method.